core/node/rules/can_add_event.go

```diff
@@ -1256,6 +1334,48 @@ func (ru *aeUnpinRules) validUnpin() (bool, error) {
 	return false, RiverError(Err_INVALID_ARGUMENT, "message is not pinned")
 }
 
+func (ru *aeUpdateAutojoinRules) validUpdateAutojoin() (bool, error) {
+	if ru.updateAutojoin == nil {
+		return false, RiverError(Err_INVALID_ARGUMENT, "event is not an update autojoin event")
+	}
+	view := ru.params.streamView.(events.SpaceStreamView)
+	channelId, err := shared.StreamIdFromBytes(ru.updateAutojoin.ChannelId)
+	if err != nil {
+		return false, err
+	}
+
+	// check if the channel exists
+	_, err = view.GetChannelInfo(channelId)
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
+}
+
+func (ru *aeUpdateChannelShowUserJoinLeaveEventsRules) validUpdateChannelShowUserJoinLeaveEvents() (bool, error) {
+	if ru.update == nil {
+		return false, RiverError(
+			Err_INVALID_ARGUMENT,
+			"event is not an update channel show user join leave events event",
+		)
+	}
+	view := ru.params.streamView.(events.SpaceStreamView)
+
+	channelId, err := shared.StreamIdFromBytes(ru.update.ChannelId)
+	if err != nil {
+		return false, err
+	}
+
+	// check if the channel exists
+	_, err = view.GetChannelInfo(channelId)
+	if err != nil {
+		return false, err
+	}
+
+	return true, nil
```

Comment:
can you create a third function called "channelExistsInSpace" and call it from these two new functions?